File: OtakuFest/OtakuFest/frmPagamento.vb
Public Class frmPagamento

    Private Sub frmPagamento_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.Venda_CompraTableAdapter.Fill(Me.Banco_OtakuFestDataSet._Venda_Compra)
        Me.ClienteTableAdapter.Fill(Me.Banco_OtakuFestDataSet.Cliente)
        Me.Pagamento_do_ClienteTableAdapter.Fill(Me.Banco_OtakuFestDataSet.Pagamento_do_Cliente)
        If Pagamento_do_ClienteBindingSource.Count = 0 Then
            Pagamento_do_ClienteBindingSource.AddNew()
            ExclusivoCheckBox.Checked = False
            Valor_FinalTextBox.Clear()
            DescontosTextBox.Clear()
            Data_da_CompraComboBox.Focus()
        End If
    End Sub

    Private Sub Nome_do_ClienteComboBox_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Nome_do_ClienteComboBox.SelectedIndexChanged
        Dim X As Integer = Nome_do_ClienteComboBox.SelectedIndex
        ID_do_ClienteComboBox.SelectedIndex = X
    End Sub

    Private Sub ExclusivoCheckBox_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ExclusivoCheckBox.CheckedChanged
        Dim Valor_Total As Decimal = Valor_TotalComboBox.SelectedValue
        Dim Descontos As Decimal
        Dim Valor_Final As Decimal

        If ExclusivoCheckBox.Checked = True Then
            Descontos = Valor_Total * 0.1
            Valor_Final = Valor_Total - Descontos
        Else
            Descontos = 0
            Valor_Final = Valor_Total - Descontos
        End If

        DescontosTextBox.Text = String.Format("{0:C2}", Descontos)
        Valor_FinalTextBox.Text = String.Format("{0:C2}", Valor_Final)
    End Sub

    Private Sub Data_da_CompraComboBox_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Data_da_CompraComboBox.SelectedIndexChanged
        Dim X As Integer = Data_da_CompraComboBox.SelectedIndex
        Valor_TotalComboBox.SelectedIndex = X

        Dim Valor_Total As Decimal = Valor_TotalComboBox.SelectedValue
        Dim Descontos As Decimal
        Dim Valor_Final As Decimal

        If ExclusivoCheckBox.Checked = True Then
            Descontos = Valor_Total * 0.1
            Valor_Final = Valor_Total - Descontos
        Else
            Descontos = 0
            Valor_Final = Valor_Total - Descontos
        End If

        DescontosTextBox.Text = String.Format("{0:C2}", Descontos)
        Valor_FinalTextBox.Text = String.Format("{0:C2}", Valor_Final)
    End Sub

    Private Sub Pagamento_do_ClienteBindingNavigatorSaveItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Pagamento_do_ClienteBindingNavigatorSaveItem.Click
        Try
            Me.Validate()
            Me.Pagamento_do_ClienteBindingSource.EndEdit()
            Me.TableAdapterManager.UpdateAll(Me.Banco_OtakuFestDataSet)
            MessageBox.Show("Registro salvo com sucesso!", "Mensagem do Sistema", MessageBoxButtons.OK, MessageBoxIcon.Information)
        Catch ex As Exception
            MessageBox.Show("Falha ao salvar o registro!", "Mensagem do Sistema", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub

    Private Sub BindingNavigatorAddNewItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorAddNewItem.Click
        Forma_de_PagamentoComboBox.SelectedIndex = -1
        Data_da_CompraComboBox.SelectedIndex = -1
        Nome_do_ClienteComboBox.SelectedIndex = -1
        ID_do_ClienteComboBox.SelectedIndex = -1
        ExclusivoCheckBox.Checked = False
        Valor_TotalComboBox.SelectedIndex = -1
        Data_do_PagamentoMaskedTextBox.Focus()
    End Sub

    Private Sub BindingNavigatorDeleteItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BindingNavigatorDeleteItem.Click
        Try
            If MessageBox.Show("Deseja excluir o registro?", "Mensagem do Sistema", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = Windows.Forms.DialogResult.Yes Then
                Pagamento_do_ClienteBindingSource.RemoveCurrent()
                Me.Validate()
                Me.Pagamento_do_ClienteBindingSource.EndEdit()
                Me.TableAdapterManager.UpdateAll(Me.Banco_OtakuFestDataSet)
            Else
                Me.Pagamento_do_ClienteTableAdapter.Fill(Me.Banco_OtakuFestDataSet.Pagamento_do_Cliente)
            End If
        Catch ex As Exception
            MessageBox.Show("Falha ao excluir o registro!", "Mensagem do Sistema", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub

    Private Sub frmPagamento_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed
        Me.Dispose()
    End Sub
End Class

File: KnowledgeWithoutBorders/System_KWB/frmPagamento.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Globalization;

namespace System_KWB
{
    public partial class frmPagamento : Form
    {
        bool Ocupado;
        Control Campo = new Control();

        static double Salario_Bruto;
        static double Descontos;
        static double Valor_Hora;
        static double Horas_Trabalhadas;
        static double Horas_Extras;
        static double Salario_Liquido;

        public frmPagamento()
        {
            InitializeComponent();
        }

        private void frmPagamento_Load(object sender, EventArgs e)
        {
            this.professoresTableAdapter.Fill(this.bD_KWB_DataSet.Professores);
            this.pagamento_do_ProfessorTableAdapter.Fill(this.bD_KWB_DataSet.Pagamento_do_Professor);

            if (pagamento_do_ProfessorBindingSource.Count == 0)
            {
                DataAdicionar.PerformClick();
            }

            descontosTextBox.Text = string.Format("{0:C2}", descontosTextBox.Text);
            salario_LiquidoTextBox.Text = string.Format("{0:C2}", salario_LiquidoTextBox.Text);
        }

        private void nome_do_ProfessorComboBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            int x = nome_do_ProfessorComboBox.SelectedIndex;
            código_do_ProfessorComboBox.SelectedIndex = x;
            conta_BancariaComboBox.SelectedIndex = x;
            agênciaComboBox.SelectedIndex = x;
            nome_da_AgênciaComboBox.SelectedIndex = x;
        }

        private void DataAdicionar_Click(object sender, EventArgs e)
        {
            if (this.pagamento_do_ProfessorBindingSource.Count == 0)
            {
                this.pagamento_do_ProfessorBindingSource.AddNew();
            }

            else if (this.pagamento_do_ProfessorBindingSource.Count > 0)
            {
                DataSalvar.PerformClick();
                this.pagamento_do_ProfessorBindingSource.AddNew();
            }

            nome_do_ProfessorComboBox.Focus();
            nome_do_ProfessorComboBox.SelectedIndex = -1;
            nome_da_AgênciaComboBox.SelectedIndex = -1;
            código_do_ProfessorComboBox.SelectedIndex = -1;
            conta_BancariaComboBox.SelectedIndex = -1;
            agênciaComboBox.SelectedIndex = -1;
            nome_da_AgênciaComboBox.SelectedIndex = -1;
            horas_TrabalhadasNumericUpDown.Value = 0;
            data_de_PagamentoDateTimePicker.Value = DateTime.Now;
        }

        private void DataRemover_Click(object sender, EventArgs e)
        {
            if (this.pagamento_do_ProfessorBindingSource.Count == 0)
            {
                MessageBox.Show("Não existem registros para serem removidos!", "Mensagem do Sistema", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }

            else if (this.pagamento_do_ProfessorBindingSource.Count > 0)
            {
                DialogResult opcaoSelecionada = MessageBox.Show("Remover Registro?", "Mensagem do Sistema", MessageBoxButtons.YesNo, MessageBoxIcon.Question);
                if (opcaoSelecionada.Equals(DialogResult.Yes))
                {
                    this.pagamento_do_ProfessorBindingSource.RemoveCurrent();
                }
                else
                {
                    this.pagamento_do_ProfessorBindingSource.Position = this.pagamento_do_ProfessorBindingSource.Position;
                    this.pagamento_do_ProfessorTableAdapter.Fill(this.bD_KWB_DataSet.Pagamento_do_Professor);
                }
            }

            nome_do_ProfessorComboBox.Focus();
        }

        private void DataSalvar_Click(object sender, EventArgs e)
        {
            if (data_de_PagamentoDateTimePicker.Text == "")
            {
                data_de_PagamentoDateTimePicker.Value = DateTime.Now;
            }

            if (this.pagamento_do_ProfessorBindingSource.Count == 0)
            {
                MessageBox.Show("Não existem registros para serem salvos!", "Mensagem do Sistema", MessageBoxButtons.OK, MessageBoxIcon.Error);
            }

            else if (this.pagamento_do_ProfessorBindingSource.Count > 0)
            {
                if (Ocupado == true)
                {
                    return;
                }

                Ocupado = true;

                this.Validate();
                this.pagamento_do_ProfessorBindingSource.EndEdit();

                try
                {
                    this.pagamento_do_ProfessorTableAdapter.Update(bD_KWB_DataSet.Pagamento_do_Professor);
                    MessageBox.Show("Salvo com Sucesso...", "Aviso", MessageBoxButtons.OK, MessageBoxIcon.Information);
                }

                catch (Exception)
                {
                    MessageBox.Show("Erro ao Salvar!");
                }

                Ocupado = false;

                this.pagamento_do_ProfessorTableAdapter.Fill(this.bD_KWB_DataSet.Pagamento_do_Professor);
                this.pagamento_do_ProfessorBindingSource.Position = this.pagamento_do_ProfessorBindingSource.Position;
            }
        }

        private void DataPrimeiro_Click(object sender, EventArgs e)
        {
            this.pagamento_do_ProfessorBindingSource.MoveFirst();
        }

        private void DataAnterior_Click(object sender, EventArgs e)
        {
            this.pagamento_do_ProfessorBindingSource.MovePrevious();
        }

        private void DataProximo_Click(object sender, EventArgs e)
        {
            this.pagamento_do_ProfessorBindingSource.MoveNext();
        }

        private void DataUltimo_Click(object sender, EventArgs e)
        {
            this.pagamento_do_ProfessorBindingSource.MoveLast();
        }

        private void salario_BrutoTextBox_Enter(object sender, EventArgs e)
        {
            salario_BrutoTextBox.Clear();
        }

        private void salario_BrutoTextBox_Leave(object sender, EventArgs e)
        {
            if (salario_BrutoTextBox.Text == "")
            {
                salario_BrutoTextBox.Text = "0";
            }
        }

        private void valor_da_HoraTextBox_Enter(object sender, EventArgs e)
        {
            valor_da_HoraTextBox.Clear();
        }

        private void valor_da_HoraTextBox_Leave(object sender, EventArgs e)
        {
            if (valor_da_HoraTextBox.Text == "")
            {
                valor_da_HoraTextBox.Text = "0";
            }
        }

        private void horas_ExtrasTextBox_Enter(object sender, EventArgs e)
        {
            horas_ExtrasTextBox.Clear();
        }

        private void horas_ExtrasTextBox_Leave(object sender, EventArgs e)
        {
            if (horas_ExtrasTextBox.Text == "")
            {
                horas_ExtrasTextBox.Text = "0";
            }
        }

        private void btnCalcular_Click(object sender, EventArgs e)
        {
            try
            {
                Valor_Hora = Convert.ToDouble(valor_da_HoraTextBox.Text);
                Horas_Trabalhadas = Convert.ToDouble(horas_TrabalhadasNumericUpDown.Value);
                Salario_Bruto = Convert.ToDouble(salario_BrutoTextBox.Text);
                Horas_Extras = Convert.ToDouble(horas_ExtrasTextBox.Text);
            }
            catch (Exception)
            {
                MessageBox.Show("Algum campo está vazio!");
            }

            try
            {
                Salario_Liquido = (Salario_Bruto + Horas_Extras + (Valor_Hora * Horas_Trabalhadas) - Descontos);
                descontosTextBox.Text = string.Format("{0:C2}", Descontos);
                salario_LiquidoTextBox.Text = string.Format("{0:C2}", Salario_Liquido);
            }
            catch(Exception)
            {
                MessageBox.Show("Algum campo possui valor inválido!");
            }
        }

        private void salario_BrutoTextBox_TextChanged(object sender, EventArgs e)
        {
            if (double.TryParse(salario_BrutoTextBox.Text, out Salario_Bruto))
            {
                Descontos = (Salario_Bruto * 0.05);
                descontosTextBox.Text = string.Format("{0:C2}", Descontos);
            }
        }

        private void btnLimpar_Click(object sender, EventArgs e)
        {
            salario_BrutoTextBox.Focus();

            salario_BrutoTextBox.Clear();
            Salario_Bruto = 0;
            descontosTextBox.Clear();
            Descontos = 0;
            horas_TrabalhadasNumericUpDown.Value = 0;
            Horas_Trabalhadas = 0;
            valor_da_HoraTextBox.Clear();
            Valor_Hora = 0;
            horas_ExtrasTextBox.Clear();
            Horas_Extras = 0;
            salario_LiquidoTextBox.Clear();
            Salario_Liquido = 0;
        }

        private void nome_do_ProfessorComboBox_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                SendKeys.Send("{TAB}");
            }
        }
    }
}
